### PR TITLE
[plugin.video.raiyoyo] 1.0.0

### PR DESCRIPTION
**Pull request by codingkoala**
### Description

A kodi plugin to access Rai YoYo website and its content, Rai YoYo is a Rai (italian TV) cartoon channel.
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by Razzeee** [comment]
your missing a fanart

File: plugin.video.raiyoyo/addon.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        <import addon="xbmc.python" version="2.24.0" />
+        <import addon="script.module.phate89" version="1.0.0"/>
+    </requires>
+    <extension library="default.py" point="xbmc.python.pluginsource">
+        <provides>video</provides>
+    </extension>
+    <extension point="xbmc.addon.metadata">
+        <language>it</language>
+        <platform>all</platform>
+        <summary lang="it">I programmi di RAI YoYo</summary>
```
Comment (Razzeee) [inline plugin.video.raiyoyo/addon.xml]: please switch to the new locales instead of using languages so it should be `it_IT` and `en_GB`

**Comment by Razzeee** [comment]
and you might want to look at http://kodi.wiki/view/Addon.xml#.3Cassets.3E

File: plugin.video.raiyoyo/addon.xml (hunk 1 above)
Comment (codingkoala) [inline plugin.video.raiyoyo/addon.xml]: Hi thanks for reviewing ... do I need to make changes and commit for approval ? 
Sorry for asking but I am new.

Comment (Razzeee) [inline plugin.video.raiyoyo/addon.xml]: no, problem, just make the changes and push them on top, we will squash them for you later

Comment (Razzeee) [inline plugin.video.raiyoyo/addon.xml]: and don't be afraid to ask, if you hit any walls

Comment (codingkoala) [inline plugin.video.raiyoyo/addon.xml]: Thank you very much!

**Comment by Razzeee** [comment]
@codingkoala your still missing a fanart see http://kodi.wiki/view/Add-on_rules#General_requirements

**Comment by codingkoala** [comment]
just pushed now fanart I hope it's ok 
Thank you

**Comment by Razzeee** [comment]
Thank you, looks good to me.

**Comment by codingkoala** [comment]
Great thanks so what's the next step? 
Thank you for your patience.

**Comment by Razzeee** [comment]
I would like another team member to have a look as this is a new addon in the repo. @enen92 @tadly

File: plugin.video.raiyoyo/addon.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="plugin.video.raiyoyo" name="RAI YoYo" provider-name="codingkoala" version="1.0.0">
+    <requires>
+        <import addon="xbmc.python" version="2.24.0" />
```
Comment (enen92) [inline plugin.video.raiyoyo/addon.xml]: 2.25.0

File: plugin.video.raiyoyo/default.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+def elenco_puntate(id):
+    kodiutils.setContent('videos')
+    for pun in raiyoyo.get_url_punList(id):
+      stamp_ep(pun)
+    kodiutils.endScript()
+
+
+def playRaiyoyo(streamUrl):
+    kodiutils.setResolvedUrl(streamUrl)
```
Comment (enen92) [inline plugin.video.raiyoyo/default.py]: Why dont you simply setResolveUrl here? You are calling the method twice and then just executing a builtin function. Use xbmc.Player().play() instead

File: plugin.video.raiyoyo/addon.xml (hunk 2 above)
Comment (codingkoala) [inline plugin.video.raiyoyo/addon.xml]: Hi enen92 sure I will update the version number!
Thanks

File: plugin.video.raiyoyo/default.py (hunk 3 above)
Comment (codingkoala) [inline plugin.video.raiyoyo/default.py]: Actually I had some issue with playing on my Mac osx Kodi 16.1 and tried to force using play() ... probably the issue was temporary and then it worked fine ... I also tested on my Krypton kodi on tablet and on Jarvis on tv box and it works fine. 
I forgot to delete the play() ... I will commit changes together with version number change later, thank you.

**Comment by Tadly** [comment]
Only other thing I have to note is that you keep mixing 4 space and 2 space indentation which is generally bad practices in python development as it's a language which heavily relies on proper indentation.
Decide on one indentation and use it throughout the whole project, be it 2 spaces, 4, spaces, 1 tab, 2 tabs etc.

IMO not needed to be changed right away but something you should keep in the back of your head.

**Comment by codingkoala** [comment]
thanks Tadly I will review the whole code on next version and be consistent on that, I am not a phyton coder actually, I do code in C/C++/PHP/Javascript, I just made this addon for my little child and it's the first time I use phyton I based on another addon code to start and probably different editor setup lead to different spaces/tabs.

Thanks for noticing that !

**Comment by codingkoala** [comment]
as requested I changed addon.xml with new version number and commented out extra lines not needed in default.py, will do a code cleanup in next version.

Thanks

**Comment by enen92** [comment]
Thanks @codingkoala for adjusting, merged. To fix the indentation on a next release as pointed by @Tadly you might want to consider the reindent script (http://svn.python.org/projects/python/trunk/Tools/scripts/reindent.py). It's just a question of running it against your python file.

Also just another comment, it's usually not a good practice to have methods with just one or two lines of code. In those cases, just execute the statements "inline" with the rest of the code (e.g: the resolveurl method). If you want to know more about that: https://en.wikipedia.org/wiki/Cyclomatic_complexity

Regards

**Comment by codingkoala** [comment]
thank you enen92 for good advices and all the other reviewers for helping me in this first release, much appreciated.

Thanks

**Pull request merged**